### ISSUE TITLE
Excessive reading from the _namespace table when initializing SweepStrategyManager

### ISSUE THREAD

**Issue by cholve** [palantir/atlasdb#1485]
When initializing the SweepStrategyManager, we go through the TableRemappingKeyValueService to get all of the appropriate table references and their metadata.  However, the way the loop is written means you can get into a situation where you re-read the entire _namespace table for every unmapped table.  If the number of entries in _namespace is large, this can be very painful.  

Here's the sad loop from TableRemappingKeyValueService.java:211
```
        for (Entry<TableReference, byte[]> entry : delegate().getMetadataForTables().entrySet()) {
            tableRefToBytes.put(
                    Iterables.getOnlyElement(tableMapper.mapToFullTableNames(ImmutableSet.of(entry.getKey()))),
                    entry.getValue());
        }
```

This is causing pain for Gotham when the number of namespaced tables is large.  We should be able return the full mapping so that we don't have to call mapToFullTableNames so many times.

**Comment by rhero** [comment]
Just to be consistent with the other comments on the PRs, we should have someone from the London team review the associated PRs. #1486 #1492

**Comment by jeremyk-91** [comment]
Fixed in #1486 and #1568.
